nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutFile.java

```diff
@@ -232,8 +240,14 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             final Path rootDirPath = configuredRootDirPath.toAbsolutePath();
             String filename = flowFile.getAttribute(CoreAttributes.FILENAME.key());
             final Path tempCopyFile = rootDirPath.resolve("." + filename);
-            final Path copyFile = rootDirPath.resolve(filename);
-
+            final Path copyFile = rootDirPath.resolve(filename)
+            if (context.getProperty(PREVENT_PATH_ESCAPE).asBoolean() && !copyFile.startsWith(rootDirPath)) {
+                flowFile = session.penalize(flowFile);
+                session.transfer(flowFile, REL_FAILURE);
+                logger.error("Resolved path escapes the root dir path");
```

Comment:
I would provide the two path values that were compared/tested so the user can understand what really happened.

Reply:
Do you mean:
!(copyFile.startsWith(rootDirPath) ||  tempCopyFile.startsWith(rootDirPath))